### ISSUE TITLE
Report Filter "Not Equals" not Excluding Input

### ISSUE THREAD

**Issue by Paul979** [mautic/mautic#3346]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? |  V
| Feature request? | 
| Enhancement? | 

## Description:
When filtering using "not equals" it functions as "equals". See screenshots below...

## If a bug:

| Q   | A
| --- | ---
| Mautic version | 2.5.1
| PHP version | 7.0

### Steps to reproduce:
1. See screenshots...

![filter-setup](https://cloud.githubusercontent.com/assets/25543761/22627215/ccba9532-ebbe-11e6-9e1a-2552ad546209.png)
![selecting-instead-of-excluding](https://cloud.githubusercontent.com/assets/25543761/22627217/d0dc1f64-ebbe-11e6-9485-53659804c3ac.png)

 
### Log errors: 

_Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable._

**Comment by Paul979** [comment]
Same is also true when filtering with categories and I assume all other filter columns.

Not equals is not working.

And also, when I set dynamic to no, it's never saved. It always stays at yes.

**Comment by Paul979** [comment]
"Not like" doesn't work also, when I enter anything when the condition is "not like" then all records are all filtered away, even when the value that I enter is not in any of those record.

What is happening here? Are these filters just not working yet? Or is this a bug with my system? It feels very broken right now the report filtering system, unless I'm missing something here.

**Comment by Paul979** [comment]
When removing a filter the filter also seems to stay applied to the report. I just had to recreate the filter, remove the filter value, and delete the filter again to make all records show up again. (this was with a regular "equals" filter which does work).

This seems like  recipy for confusion and frustration. It can become impossble to troubleshoot when a deleted filter is still filtering your report.

**Comment by Paul979** [comment]
Something similar also happened when I temporarily set up a Segment Filter...

Here's what happened:

1. I set up a segment filter that targeted only 1 specific contact.
2. The contact was added to that segment by the filter.
3. I'm not sure. I either :
A) removed the filter which then automatically removed the contact from the segment, or...
B) I manually removed that contact from the segment. 
It was probably A, and I definitly removed that filter in any case.
4. Then in Contact Widgets>Top Segments that 1 contact was still showing in that temporary segment, even though it had been removed from that segment (checking the segment the segment menu, it showed No Contacts).
5. I then re-added the contact in question to the segment, and then right after removed it from the segment again. And then finally it was completely removed from the segment, and not showing up in the Top Segments widget anymore.
